tests/acceptance/30_custom_promise_types/24_expireafter.cf

```diff
@@ -0,0 +1,98 @@
+##############################################################################
+#
+#  Test that expireafter works with custom promise types.
+#
+##############################################################################
+
+body common control
+{
+  inputs => { "../default.cf.sub" };
+  bundlesequence  => { default("$(this.promise_filename)") };
+  version => "1.0";
+}
+
+bundle common version_check
+{
+  classes:
+      "compatible_python_version"
+        expression => returnszero(
+          "/usr/bin/python3 -c 'import sys; assert sys.version_info >= (3,6)'",
+          "useshell"
+        );
+
+@if minimum_version(3.22)
```

Comment:
How do we know this will work in 3.22?

Reply:
Well, the failing test will at least punish us for not fixing things in time! :+1:

Reply:
Just for commentary, you could have used the `minimum_version()` function for example, but I don't think there is anything wrong with the use of the macro.

```cfengine3
bundle agent __main__
{
  classes:
      "supports_expireafter"
        expression => cf_version_minimum( "3.22" );
}
```